lib/datadog/appsec/contrib/faraday/middleware.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module Contrib
+      module Faraday
+        # Middleware for Faraday
+        class Middleware < ::Faraday::Middleware
+          def call(request_env)
+            context = AppSec.active_context
+
+            if context && AppSec.rasp_enabled?
```

Comment:
I think it's better with guard clause than nesting

Reply:
I decided against it because with a guard clause we have to write the call `@app.call(request_env)` twice - once in the guard clause, and the second time at the end of this method. What do you think, should I still do the guard clause?

Reply:
The main point - I don't need to keep in mind the context of the condition, I will focus on the main part which is - running rasp.

If I will outline it

```ruby
def call(...)
  context = ""

  if context && AppSec.rasp?
    result = ""
    if result.match?
      handle(...)
    end
  end

  @app.call(..)
end
```

versus

```ruby
def call(...)
  context = ""
  return @app.call(..) if context && AppSec.rasp?
  
  result = ""
  
  if result.match?
      handle(...)
    end
  end

  @app.call(..)
end
```